src/m3ninx/index/batch.go

```diff
@@ -138,6 +141,16 @@ func (e *BatchPartialError) Add(err BatchError) {
 	e.errs = append(e.errs, err)
 }
 
+// AddWithLock adds an error to e with a lock. Any nil errors are ignored.
+func (e *BatchPartialError) AddWithLock(err BatchError) {
+	e.Lock()
+	defer e.Unlock()
+	if err.Err == nil {
+		return
+	}
+	e.errs = append(e.errs, err)
```

Comment:
nit: can probably be a little bit more efficient as:

```golang
if err.Err == nil {
 return
}

e.Lock()
e.errs = append(e.errs, err)
e.Unlock()
```

Also, seems to only be used in one place; might be more straightforward to do the locking there rather than adding a mutex on the struct? Take it or leave it though

Reply:
Just moved the locking code around.